### ISSUE TITLE
Implementing logaddexp, logaddexp2

### ISSUE THREAD

**Issue by cloudhan** [pytorch/pytorch#38377]
I'd like implement logaddexp for cpu and cuda mentioned in #38349. these functions are perticularly useful when doing long probability accumulation and maintaining good numerical precesion. 

cc @mruberry

**Comment by mruberry** [comment]
You don't need to file an issue just work on something ;) But if you have questions or there's an issue please ask here. 

It'd be great to have logaddexp and logaddexp2.

**Comment by vadimkantorov** [comment]
Isn't logaddexp just a synonym for logsumexp. PR mentions that logsumexp doesn't support global reduction. But in that case it'd be better to add that support to logsumexp and make logaddexp just another name for logsumexp.

Some related logsumexp issues: https://github.com/pytorch/pytorch/issues/31837, https://github.com/pytorch/pytorch/issues/31829

A more general question: should PyTorch have several names for an op? (in this case of existing PyTorch method and discovering another name in NumPy) Should two names co-exist (and partially confuse users)? Should only PyTorch name be preserved? Should PyTorch name be deprecated?

**Comment by cloudhan** [comment]
> This op should be disambiguated with logsumexp which do a reduction on a tensor over a specific axis.

**Comment by cloudhan** [comment]
But I also suspect whether should I add this op. Since to use similar `logsumexp`, all you need is concat 2 tensors and do the reduction over the axis.

So for now, I might leave this here, until someone point me out the significance of this op. Otherwise feel free to close this in the future.

**Comment by mruberry** [comment]
They're two different ops. Please read the documentation. Even their signatures are different: logsumexp ([documentation](https://pytorch.org/docs/master/generated/torch.logsumexp.html?highlight=logsumexp#torch.logsumexp)) is a reduction and logaddexp is a binary pointwise operation.

**Comment by vadimkantorov** [comment]
Sorry for confusion. @cloudhan @mruberry You are right
